Sources/Apollo/GraphQLFirstReceivedAtTracker.swift

```diff
@@ -0,0 +1,38 @@
+import ApolloAPI
+import Foundation
+
+final class GraphQLFirstReceivedAtTracker: GraphQLResultAccumulator {
+
+  func accept(childObject: Date, info: FieldExecutionInfo) throws -> Date {
+    return childObject
+  }
+
+  func accept(scalar: JSONValue, firstReceivedAt: Date, info: FieldExecutionInfo) throws -> Date {
+    return firstReceivedAt
+  }
+
+  func acceptNullValue(firstReceivedAt: Date, info: FieldExecutionInfo) throws -> Date {
+    return firstReceivedAt
+  }
+
+  func acceptMissingValue(firstReceivedAt: Date, info: FieldExecutionInfo) throws -> Date {
+    return firstReceivedAt
+  }
+
+  func accept(list: [Date], info: FieldExecutionInfo) throws -> Date {
+    return list.min() ?? .distantPast
+  }
+
+  func accept(fieldEntry: Date, info: FieldExecutionInfo) throws -> Date? {
+    return fieldEntry
+  }
+
+  func accept(fieldEntries: [Date], info: ObjectExecutionInfo) throws -> Date {
+    return fieldEntries.min() ?? .distantPast
```

Comment:
❗ You'll want to make sure you're not making the same bug mistakes I made https://github.com/peek-travel/apollo-ios/pull/7/files

By doing `min` we end up with the oldest date which always ends up being `distantPast`

Reply:
added